pkg/utils/imagevector/imagevector.go

```diff
@@ -14,73 +14,81 @@
 
 package imagevector
 
-// ReadImageVector reads the image vector yaml file in the charts directory, unmarshals the content
 import (
 	"fmt"
-	"io/ioutil"
+	"io"
 	"os"
-	"path/filepath"
 	"strings"
 
-	"github.com/gardener/gardener/pkg/operation/common"
 	"github.com/gardener/gardener/pkg/utils"
-	yaml "gopkg.in/yaml.v2"
+	"gopkg.in/yaml.v2"
 )
 
-// ReadImageVector reads the image.yaml in the chart directory, unmarshals it
-// into a []*ImageSource type and returns it.
-func ReadImageVector() (ImageVector, error) {
-	path := filepath.Join(common.ChartPath, "images.yaml")
+// OverrideEnv is the name of the image vector override environment variable.
+const OverrideEnv = "IMAGEVECTOR_OVERWRITE"
 
-	vector, err := readImageVector(path)
-	if err != nil {
-		return nil, err
-	}
+// Read reads an ImageVector from the given io.Reader.
+func Read(r io.Reader) (ImageVector, error) {
+	vector := struct {
+		Images ImageVector `json:"images" yaml:"images"`
+	}{}
 
-	overwritePath := os.Getenv("IMAGEVECTOR_OVERWRITE")
-	if len(overwritePath) == 0 {
-		return vector, nil
+	if err := yaml.NewDecoder(r).Decode(&vector); err != nil {
+		return nil, err
 	}
+	return vector.Images, nil
+}
 
-	overwrite, err := readImageVector(overwritePath)
+// ReadFile reads an ImageVector from the file with the given name.
+func ReadFile(name string) (ImageVector, error) {
+	file, err := os.Open(name)
 	if err != nil {
 		return nil, err
 	}
+	defer file.Close()
 
-	overwrittenImages := make(map[string]*ImageSource, len(overwrite))
-	for _, image := range overwrite {
-		overwrittenImages[image.Name] = image
-	}
+	return Read(file)
+}
 
-	var out ImageVector
-	for _, image := range vector {
-		if overwritten, ok := overwrittenImages[image.Name]; ok {
-			if len(overwritten.Tag) == 0 && len(image.Tag) > 0 {
```

Comment:
Why is that deleted?

Reply:
It is not deleted it is moved to `Merge`

Reply:
I don't see it. Where is the special handling for the `Tag` property?